clusterdock/topologies/cdh/actions.py

```diff
@@ -77,13 +100,26 @@ def start(args):
                             **{'volumes_from': [java_image]} if args.java else {})
                        for hostname in args.secondary_nodes]
 
+    if args.kerberos:
+        kdc_node = Node(hostname=KDC_HOST_NAME, network=args.network,
+                        image=kdc_node_image, volumes=[{KERBEROS_VOLUME_DIR: KERBEROS_VOLUME_DIR}])
+
     secondary_node_group = NodeGroup(name='secondary', nodes=secondary_nodes)
     node_groups = [NodeGroup(name='primary', nodes=[primary_node]),
                    secondary_node_group]
+    if args.kerberos:
+        node_groups.append(NodeGroup(name=KDC_HOST_NAME, nodes=[kdc_node]))
 
     cluster = Cluster(topology='cdh', node_groups=node_groups, network_name=args.network)
     cluster.start()
 
+    # Cloudera cluster nodes - these exclude kdc node
+    cc_nodes = [primary_node] + secondary_nodes
+
+    if args.kerberos:
+        _install_kdc_server(cluster, kdc_node, args.network, args.kerberos_principals)
+        _install_kerberos_libs(cluster, primary_node, cc_nodes)
```

Comment:
Same comment as the above line. I think you could probably infer everything you need to know about which nodes to pick just by having a function get a `Cluster` instance and introspecting on it.